engine/kv_engine.hpp

```diff
@@ -114,6 +114,8 @@ class KVEngine : public Engine {
     return skiplists_;
   };
 
+  const std::shared_ptr<HashTable> GetHashTable() { return hash_table_; }
```

Comment:
Don't expose internal components. Use special macros to expose them for test builds.
Also, no need to return const.